packages/block-library/src/navigation-link/edit.js

```diff
@@ -133,6 +133,82 @@ function getSuggestionsQuery( type, kind ) {
 	}
 }
 
+/**
+ * @typedef {'post-type'|'custom'|'taxonomy'} WPNavigationLinkKind
```

Comment:
FYI that for nav_menu_item [`type` can also be `post-type-archive`](https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L862). As `block.attributes.kind` maps to `menuItem.type` we should probably include that here.

Reply:
Good catch updated in https://github.com/WordPress/gutenberg/pull/31259/commits/c6a8a5984af5ec1f26af1fe61b9d7ebd698eacb3